File: Sample/sample_cs/Graphics/2D/ImagePackage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sample_cs.Graphics._2D
{
	/// <summary>
	/// 画像パッケージから画像を配置する
	/// </summary>
	class ImagePackage : ISample
	{
		public void Run()
		{
			// aceを初期化する
			asd.Engine.Initialize("ImagePackage", 640, 480, new asd.EngineOption());

			// シーンを生成する
			var scene = new asd.Scene();

			// レイヤーを生成する
			var layer = new asd.Layer2D();

			// シーンにレイヤーを追加する
			scene.AddLayer(layer);

			// シーンを切り替える
			asd.Engine.ChangeScene(scene);

			// イメージパッケージを読み込む
			asd.ImagePackage imagePackage = asd.Engine.Graphics.CreateImagePackage("Data/ImagePackage/Game.aip");

			for(int i = 0; i < imagePackage.ImageCount; i++)
			{
				// テクスチャを取り出す
				asd.Texture2D texture = imagePackage.GetImage(i);
				asd.RectI area = imagePackage.GetImageArea(i);

				// テクスチャをオブジェクトとして追加する
				asd.TextureObject2D textureObject2D = new asd.TextureObject2D();
				textureObject2D.Texture = texture;
				textureObject2D.Position = new asd.Vector2DF(area.X, area.Y);
				layer.AddObject(textureObject2D);
			}

			// aceが進行可能かチェックする。
			while (asd.Engine.DoEvents())
			{
				// aceを更新する。
				asd.Engine.Update();
			}

			// aceを終了する。
			asd.Engine.Terminate();
		}
	}
}

File: Sample/BasicSample/sample_cpp/Graphics/2D/ImagePackage.cpp

#include <ace.h>

/**
@brief	ImagePackageのサンプル。画像パッケージを元に画像を配置する。
*/
void ImagePackage()
{
	// AC-Engineを初期化する。
	ace::Engine::Initialize(ace::ToAString("ImagePackage").c_str(), 640, 480, ace::EngineOption());

	// シーンを生成する
	auto scene = std::make_shared<ace::Scene>();

	// レイヤーを生成する
	auto layer = std::make_shared<ace::Layer2D>();

	// シーンにレイヤーを追加する
	scene->AddLayer(layer);

	// シーンを切り替える
	ace::Engine::ChangeScene(scene);

	// イメージパッケージを読み込む
	auto imagePackage = ace::Engine::GetGraphics()->CreateImagePackage(ace::ToAString("Data/ImagePackage/Game.aip").c_str());

	for (int i = 0; i < imagePackage->GetImageCount(); i++)
	{
		// テクスチャを取り出す
		auto texture = imagePackage->GetImage(i);
		auto area = imagePackage->GetImageArea(i);

		// テクスチャをオブジェクトとして追加する
		auto textureObject2D = std::make_shared<ace::TextureObject2D>();
		textureObject2D->SetTexture(texture);
		textureObject2D->SetPosition(ace::Vector2DF(area.X, area.Y));
		layer->AddObject(textureObject2D);
	}

	// AC-Engineのウインドウが閉じられていないか確認する。
	while (ace::Engine::DoEvents())
	{
		// AC-Engineを更新する。
		ace::Engine::Update();
	}

	// AC-Engineの終了処理をする。
	ace::Engine::Terminate();
}